src/Translator.php

```diff
@@ -150,82 +152,96 @@ protected function loadLanguageStrings($locale, $platform = null, $force = false
+                ],
+            ]);
 
-    protected function getAuthentication()
-    {
-        return [
-            $this->getUsername(),
-            $this->getPassword(),
-        ];
+            if ($res->getStatusCode() != 200) {
+                throw new ApiException("API returned status [{$res->getStatusCode()}].");
+            }
+
+            $languages = json_decode((string)$res->getBody());
+
+            Cache::forever('languagecenter.languages', $languages);
+            Cache::forever('languagecenter.timestamp', $timestamp);
+        } catch (\Exception $exception) {
+            $languages = Cache::get('languagecenter.languages');
+
+            if (is_null($languages)) {
+                throw $exception; // Nothing to do, can not restore data, throw exception
+            }
+        }
     }
 
-    protected function loadLanguages()
+    public function updateStrings($locale, $platform = null)
     {
-        $client = $this->getClient();
+        try {
+            $timestamp = Carbon::now()->timestamp;
 
-        $res = $client->request('GET', $this->getApiUrl().'languages', [
-            'auth' => $this->getAuthentication(),
-        ]);
+            if (is_null($platform)) {
+                $platform = $this->getDefaultPlatform();
+            }
 
-        if ($res->getStatusCode() != 200) {
-            throw new ApiException("API returned status [{$res->getStatusCode()}].");
-        }
+            $client = $this->getClient();
 
-        $languages = json_decode((string) $res->getBody());
+            $res = $client->request('GET', $this->getApiUrl() . 'strings?platform=' . $platform . '&language=' . $locale, [
+                'auth' => $this->getAuthentication(),
+            ]);
 
-        foreach ($languages as $language) {
-            $this->languages[] = $language->codename;
-            if ($language->is_fallback) {
-                Config::set('app.locale', $language->codename);
-                Config::set('app.fallback_locale', $language->codename);
-                $this->locale = $language->codename;
+            if ($res->getStatusCode() != 200) {
+                throw new ApiException("API returned status [{$res->getStatusCode()}].");
+            }
+
+            $strings = json_decode((string)$res->getBody());
+
+            if (!isset($this->strings[$locale])) {
+                $this->strings[$locale] = [];
             }
+
+            $this->strings[$locale][$platform] = [];
+
+            foreach ($strings as $string) {
+                $this->strings[$locale][$platform][$string->key] = $string->value;
+                $this->strings[$string->language][$platform][$string->key] = $string->value;
+            }
+
+            Cache::forever('languagecenter.strings', $this->strings);
+            Cache::forever('languagecenter.language.' . $locale . '.timestamp', $timestamp);
+        } catch (\Exception $exception) {
+            // failed to update string - it's okay - we do it later
```

Comment:
It should also work as a standalone package without Bugsnag implementation. I could however call a event which Bugnsag could catch if needed.